tests/contrib/django/django_tests.py

```diff
@@ -404,28 +446,8 @@ def test_record_none_exc_info(self):
         assert 'exception' not in event
 
     def test_404_middleware(self):
-        with self.settings(MIDDLEWARE_CLASSES=['elasticapm.contrib.django.middleware.Catch404Middleware']):
-            resp = self.client.get('/non-existant-page')
-            self.assertEquals(resp.status_code, 404)
-
-            self.assertEquals(len(self.elasticapm_client.events), 1)
-            event = self.elasticapm_client.events.pop(0)['errors'][0]
-
-            self.assertEquals(event['log']['level'], 'info')
-            self.assertEquals(event['log']['logger_name'], 'http404')
-
-            self.assertTrue('request' in event['context'])
-            request = event['context']['request']
-            self.assertEquals(request['url']['raw'], u'http://testserver/non-existant-page')
-            self.assertEquals(request['method'], 'GET')
-            self.assertEquals(request['url']['search'], '')
-            self.assertEquals(request['body'], None)
-
-    @pytest.mark.skipif(django.VERSION < (1, 10),
-                        reason='new-style middlewares')
-    def test_404_new_style_middleware(self):
-        with self.settings(MIDDLEWARE_CLASSES=None, MIDDLEWARE=[
-                'elasticapm.contrib.django.middleware.Catch404Middleware']):
+        with self.settings(**middleware_setting(django.VERSION,
+                                                ['elasticapm.contrib.django.middleware.Catch404Middleware'])):
```

Comment:
Could you briefly motivate why you removed the specific checks?

Reply:
they are all still there. I merged two duplicate test cases (one for `MIDDLEWARE_CLASSES`, one for `MIDDLEWARE`) into one, using the `middleware_setting` helper function. If you expand the code block right under this diff, you'll see the exact same lines that have been removed again.